pkg/compiler/validators/bindings.go

```diff
@@ -15,12 +15,12 @@ func validateBinding(w c.WorkflowBuilder, nodeID c.NodeID, nodeParam string, bin
 	expectedType *flyte.LiteralType, errs errors.CompileErrors) (
 	resolvedType *flyte.LiteralType, upstreamNodes []c.NodeID, ok bool) {
 
-	switch binding.GetValue().(type) {
+	switch val := binding.GetValue().(type) {
 	case *flyte.BindingData_Collection:
 		if expectedType.GetCollectionType() != nil {
-			allNodeIds := make([]c.NodeID, 0, len(binding.GetMap().GetBindings()))
+			allNodeIds := make([]c.NodeID, 0, len(val.Collection.GetBindings()))
```

Comment:
i think we should check `if val.GetCollection() != nil`

Reply:
I can add it... I don't think it adds value though... these objects are parsed protos. If this is pure go code, I would say we should check because val.Collection may have the interface binding but not the value set. Added for being extra defensive :-)